packages/js/plugins/ethereum/src/Connection.ts

```diff
@@ -0,0 +1,148 @@
+import { Signer, ethers } from "ethers";
+import {
+  ExternalProvider,
+  JsonRpcProvider,
+  WebSocketProvider,
+  Web3Provider,
+  Networkish,
+} from "@ethersproject/providers";
+import { getAddress } from "@ethersproject/address";
+
+export type Address = string;
+export type AccountIndex = number;
+export type EthereumSigner = Signer | Address | AccountIndex;
+export type EthereumProvider = string | ExternalProvider | JsonRpcProvider;
+export type EthereumClient = JsonRpcProvider;
+
+export interface ConnectionConfig {
+  provider: EthereumProvider;
+  signer?: EthereumSigner;
+}
+
+export interface ConnectionConfigs {
+  [network: string]: ConnectionConfig;
+}
+
+export interface Connections {
+  [network: string]: Connection;
+}
+
+export class Connection {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore: initialized within setProvider
+  private _client: EthereumClient;
+
+  constructor(private _config: ConnectionConfig) {
+    const { provider, signer } = _config;
+
+    // Sanitize Provider & Signer
+    this.setProvider(provider, signer !== undefined ? signer : 0);
+  }
+
+  static fromConfigs(configs: ConnectionConfigs): Connections {
+    const connections: Connections = {};
+
+    for (const network of Object.keys(configs)) {
+      connections[network] = new Connection(configs[network]);
+    }
+
+    return connections;
+  }
+
+  static fromNetwork(networkish: Networkish): Connection {
+    return new Connection({
+      provider: ethers.providers.getDefaultProvider(
+        ethers.providers.getNetwork(networkish)
+      ) as JsonRpcProvider,
+    });
+  }
```

Comment:
Isn't it possible that the networkish value is a websocket URI? Because if that's the case, `getDefaultProvider` will return a `WebsocketProvider` instead of a `JsonRpcProvider` and we may have unexpected bugs, no?

Reply:
`WebsocketProvider` derives from `JsonRpcProvider`, so this shouldn't be an issue: https://github.com/ethers-io/ethers.js/blob/3b1d3fcee6bfb5178861e26ff1a1e9daa0663ec9/packages/providers/src.ts/websocket-provider.ts#L46

Reply:
Also note that `Networkish` isn't supposed to be a URI string, but rather a network name as a string ("mainnet", "rinkeby", etc).